app/views/bamposts/_form.html.erb

```diff
@@ -0,0 +1,32 @@
+<%= form_with(model: bampost) do |form| %>
+  <% if bampost.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(bampost.errors.count, "error") %> prohibited this bampost from being saved:</h2>
```

Comment:
At the moment it doesn't allow me to create posts, raising "image can't be blank" error  - it shows 'no file chosen' even when I try to upload different image formats - jpg, jpeg or png - not sure whether this happens due to acceptable image size or format criteria that you set somewhere in the app. I couldn't find where you defined these. Maybe double-check these? 
On a more positive note, your error messages are quite descriptive and your error message formatting looks pro!